polar-core/src/formatting.rs

```diff
@@ -312,6 +312,16 @@ pub mod display {
         }
     }
 
+    impl Rule {
+        pub fn head_as_string(&self) -> String {
+            format!(
+                "{}({})",
+                self.name.to_polar(),
+                format_params(&self.params, ", ")
+            )
+        }
+    }
+
     impl fmt::Display for Rule {
         fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
```

Comment:
I think this function should shell out to `Rule::to_polar`:

```rs
    impl fmt::Display for Rule {
        fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
            write!(fmt, "{}", self.to_polar())
        }
    }
```

and we should make the formatting updates in that function

Reply:
refactored to use `to_polar` and DRY'd up `to_polar` with the `head_as_string` method.